src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java

```diff
@@ -32,16 +57,48 @@ public static boolean isValidModuleClassName(String test) { // TODO: Ensure that
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns an immutable sessions list, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public List<Session> getSessions() {
+        return Collections.unmodifiableList(sessions);
+    }
+
+    /**
+     * Returns true if both modules have the same name and session list.
+     * This defines a stronger notion of equality between two module classes.
+     *
+     * @param other the object to be compared to.
+     * @return true if objects are equal.
+     */
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof ModuleClass // instanceof handles nulls
-                && className.equals(((ModuleClass) other).className)); // state check
+                && className.equals(((ModuleClass) other).className)
+                && sessions.equals(((ModuleClass) other).sessions));
```

Comment:
Since `sessions` is a List object, should we have check the individual items of the list instead?

Reply:
> Since `sessions` is a List object, should we have check the individual items of the list instead?

iirc `List#equals()` already checks if all corresponding pairs of elements in the two lists are equal so as long as `Session` overrides `equals`, it should work